waitforx/waitforx.c

```diff
@@ -0,0 +1,96 @@
+#include <X11/extensions/Xrandr.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/signal.h>
+#include <unistd.h>
+
+#define ATTEMPTS 10
+#define ALARM_WAIT 30
+
+void
+alarm_handler(int signal_num)
+{
+    printf("Unable to find RandR outputs after %d seconds\n", ALARM_WAIT);
+    exit(1);
+}
+
+int
+main(int argc, char **argv)
+{
+    char *display = NULL;
+    int error_base = 0;
+    int event_base = 0;
+    int n = 0;
+    int outputs = 0;
+    int wait = ATTEMPTS;
+
+    Display *dpy = NULL;
+    XRRScreenResources *res = NULL;
+
+    display = getenv("DISPLAY");
+
+    signal(SIGALRM, alarm_handler);
+    alarm(ALARM_WAIT);
+
+    if (!display)
+    {
+        printf("DISPLAY is null");
+        exit(1);
+    }
+
+    for (n = 1; n <= wait; ++n)
+    {
+        dpy = XOpenDisplay(display);
+        printf("Opening display %s. Attempt %d of %d\n", display, n, wait);
+        if (dpy != NULL)
+        {
+            printf("Opened display %s\n", display);
+            break;
+        }
+        sleep(1);
```

Comment:
`sleep()` isn't a standard C function - can we use `g_sleep(1000)` here?

Reply:
See d7d0866